ckcp/gitops.sh

```diff
@@ -0,0 +1,340 @@
+  # Post install
+  #############################################################################
+  # Copy the kubeconfig of kcp from inside the pod onto the local filesystem
+  if [ "$kube_dir/.gitignore" -nt "$KUBECONFIG_KCP" ]; then
+    oc cp ""$APP"/$podname:/workspace/.kcp/admin.kubeconfig" "$KUBECONFIG_KCP" >/dev/null
+  fi
+
+  # Check if external ip is assigned and replace kcp's external IP in the kubeconfig file
+  echo -n "  - External IP: "
+  if grep -q "localhost" "$KUBECONFIG_KCP"; then
+    while [ -z "$(oc get service ckcp-service -n "$ns" -o jsonpath='{.status.loadBalancer.ingress[0]}')" ]; do
+      echo -n "."
+      sleep 2
+    done
+
+    local external_ip
+    external_ip=$(oc get service ckcp-service -n "$APP" -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+    external_ip+=$(oc get service ckcp-service -n "$APP" -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    sed -i "s/localhost/$external_ip/g" "$KUBECONFIG_KCP"
+  fi
+  echo "OK"
+
+  # Make sure access to kcp-in-a-pod is good
+  echo -n "  - KCP api-server: "
+  KUBECONFIG="$KUBECONFIG_KCP" oc config set-cluster admin --insecure-skip-tls-verify=true >/dev/null
+  test_cmd=""
+  until KUBECONFIG="$KUBECONFIG_KCP" oc api-resources >/dev/null 2>&1; do
+    echo -n "."
+    sleep 2
+  done
+  echo "OK"
+
+  # Create secret
+  echo -n "  - Register KCP secret to host cluster: "
+  if ! oc get namespace pipelines >/dev/null 2>&1; then
+    echo -n "."
+    oc create namespace pipelines >/dev/null
+  fi
+  if ! oc get secret ckcp-kubeconfig -n pipelines >/dev/null 2>&1; then
+    echo -n "."
+    oc create secret generic ckcp-kubeconfig -n pipelines --from-file "$KUBECONFIG_KCP" >/dev/null
+  fi
+  echo "OK"
+
+  # Register the host cluster to KCP
+  echo -n "  - Cluster registration: "
+  if ! KUBECONFIG=$KUBECONFIG_KCP oc get cluster local >/dev/null 2>&1; then
+    (curl --fail --silent https://raw.githubusercontent.com/kcp-dev/kcp/main/contrib/examples/cluster.yaml; sed -e 's/^/    /' $KUBECONFIG) > "$kube_dir/cluster.yaml"
+    KUBECONFIG="$KUBECONFIG_KCP" oc apply -f "$kube_dir/cluster.yaml" >/dev/null
+    rm "$kube_dir/cluster.yaml"
+  fi
+  echo "OK"
+
+  # Register the KCP cluster into ArgoCD
+  export KUBECONFIG="$KUBECONFIG_MERGED"
+  echo -n "  - ArgoCD KCP registration: "
+  if ! argocd cluster get kcp >/dev/null 2>&1; then
+
+    # <workaround> See https://github.com/kcp-dev/kcp/issues/535
+    # Manually create the serviceaccount and secret on the KCP cluster for ArgoCD to use
+    local argocd_yaml="$kube_dir/argocd_kcp.yaml"
+    local argocd_sa="argocd-manager"
+    local argocd_ns="kube-system"
+    echo "---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: $argocd_sa
+  namespace: $argocd_ns
+  annotations:
+    kubernetes.io/service-account.name: $argocd_sa
+type: kubernetes.io/service-account-token
+data:
+  ca.crt: $(cat "$KUBECONFIG_KCP" | yq ".clusters.[0].cluster.certificate-authority-data")
```

Comment:
```
  - ArgoCD KCP registration: Error: unknown command ".clusters.[0].cluster.certificate-authority-data" for "yq"
Run 'yq --help' for usage.
Error: unknown command ".users.[0].user.token" for "yq"
Run 'yq --help' for usage.
```

Reply:
Can you ensure you have this version of yq installed (http://mikefarah.github.io/yq/#install).

Reply:
Installed, but now with new cluster it's working.